File: code/2419.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int target = *max_element(nums.begin(), nums.end());
        int streak = 0;
        int best = 1;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == target) {
                streak++;
                best = max(best, streak);
            }
            else {
                streak = 0;
            }
        }
        return best;        
    }
};

File: code/2419.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        best = max(nums)
        ans, streak = 0, 0
        for num in nums:
            if num == best:
                streak += 1
            else:
                streak = 0

            ans = max(ans, streak)
        return ans
